### ISSUE TITLE
Feature Request: Unique keys combination

### ISSUE THREAD

**Issue by AmirTugi** [typeorm/typeorm#72]
Hi,
I have a scenario where I want to use 2 fields as a unique key (so the combination of them both would be unique). Just like [here](http://dba.stackexchange.com/questions/17761/how-to-use-unique-key-via-combinations-of-table-fields).
Currently, I see no implementation of this in TypeORM, and I think it'd be a nice-to-have feature.

**Comment by pleerock** [comment]
Hi,
you can use [indices](https://typeorm.github.io/indices.html):


```typescript
import {Table, Index, PrimaryGeneratedColumn, Column} from "typeorm";

@Table()
@Index("my_index_with_id_and_text", ["id", "text"], { unique: true }) // first style
@Index("my_index_with_id_and_title", (post: Post) => [post.id, post.title], { unique: true }) // alternative, type-safe style
export class Post {

    @PrimaryGeneratedColumn()
    id: number;

    @Column()
    title: string;

    @Column()
    text: string;

    @Column()
    likesCount: number;

}
```

**Comment by AmirTugi** [comment]
Mmm, nope, not working

```
import {Table, Column, PrimaryGeneratedColumn, ManyToOne, CreateDateColumn, UpdateDateColumn, Index} from "typeorm";
import {Item} from "./item";

@Table()
@Index("index_item_sequence", ['item', 'sequence'], { unique: true })
export class Attachment {
    @PrimaryGeneratedColumn()
    id: number;
    @Column()
    path: string = '';
    @Column()
    isPublic: boolean = true;
    @Column()
    type: string = 'attachment';
    @Column()
    size: number = 0;
    @Column()
    link: string = '';
    @Column()
    sequence: number = 1;
    // Automatically insert, update, and remove attachments attached to the item.s
    @ManyToOne(type => Item, item => item.attachments, { cascadeAll: true })
    item: Item;
    @Column()
    comments: string = '';
    @CreateDateColumn()
    createdAt: string;
    @UpdateDateColumn()
    updatedAt: string;

}
```

I can still create an attachment with the same sequence for a specific item.

**Comment by pleerock** [comment]
hmm it should work, can you enable sql logging and check what sql it generates? it should create an index with unique properties item+sequence

**Comment by AmirTugi** [comment]
Ok, I did a small research, and after figuring that the the index was created, but not uniquely, I decided to dive into the code, and look for the problem.

I found that in `/decorator/index.js` there's a bug in line 9:
`const options = typeof maybeFieldsOrOptions === "object" ? maybeFieldsOrOptions : maybeOptions;`

From looking in the documentation of JavaScript, an array is an object (Lol), and thus the unique IndexOptions was dropped.

A quick fix is to add the `Array.isArray()` to assure that it's an object, yet not an array.

`const options = (typeof maybeFieldsOrOptions === "object" && !Array.isArray(maybeFieldsOrOptions)) ? maybeFieldsOrOptions : maybeOptions;`

After I fixed that line, it worked perfectly.
